DataBindings/Botify
# Feature: Retrieve Spotify token for API Access

## Description

This PR introduces a function `getAccessToken` that retrieves a Spotify access token using client credentials and a refresh token. It makes a POST request to the Spotify Accounts API to obtain a new access token, handles errors, and returns the access token data if successful. Additionally, it includes test cases for various scenarios.

## Changes Made

- Added the `getAccessToken` function that retrieves Spotify access tokens.
- Created unit tests for the `getAccessToken` function to cover different scenarios.
- Included error handling for invalid client credentials and refresh tokens.

## Motivation and Context

This PR aims to enhance the codebase by providing a reusable function for obtaining Spotify access tokens. It also improves code quality by adding test cases to ensure the function behaves correctly.

## How to Test

1. Ensure you have the required environment variables (`SPOTIFY_CLIENT_ID`, `SPOTIFY_CLIENT_SECRET`, `SPOTIFY_REFRESH_TOKEN`) set correctly.
2. Run the test suite using a testing framework (e.g., Jest) to verify the functionality of the `getAccessToken` function.

## Screenshots (if applicable)

N/A

## Checklist

- [x] The code builds and runs successfully.
- [x] Unit tests cover different scenarios.
- [x] Error handling is in place for invalid client credentials and refresh tokens.
- [x] The code adheres to the project's coding standards.
- [x] Documentation and comments are added as necessary.

## Related Issues

N/A
